javaslang-test/src/main/java/javaslang/test/Gen.java

```diff
@@ -424,3 +413,27 @@
         return f.apply(this);
     }
 }
+
+final class GenModule {
+
+    /**
+     * Chooses a Gen according to the given frequencies.
+     *
+     * @param n    a random value between 0 and sum(frequencies) - 1
+     * @param iter a non-empty Iterator of (frequency, Gen) pairs
+     * @param <T>  type of generated values
+     * @return A value generator, choosen according to the given frequencies and the underlying n
+     */
+    static <T> Gen<T> frequency(int n, java.util.Iterator<Tuple2<Integer, Gen<T>>> iter) {
+        do {
+            final Tuple2<Integer, Gen<T>> freqGen = iter.next();
+            final int k = freqGen._1;
+            if (n <= k) {
+                return freqGen._2;
+            } else {
+                n = n - k;
```

Comment:
Ok for me - it is performed on a copy on the stack. Btw - when dealing with immutable objects (like in Javaslang), then nothing evil can happen at all!